lib/rust/prelude/src/data/non_empty_vec.rs

```diff
@@ -288,12 +309,14 @@ impl<T> NonEmptyVec<T> {
     /// assert_eq!(drained, [1, 2, 3, 4, 5])
     /// ```
     pub fn drain<R>(&mut self, range: R) -> Drain<T>
-    where R: RangeBounds<usize> {
-        if range.contains(&0) {
+    where
+        R: RangeBounds<I>,
+        I: PartialOrd + Copy + From<usize>, {
```

Comment:
Comment above about the `From` bound applies here too

Reply:
which comment?

Reply:
`From<u8>` would make this more general